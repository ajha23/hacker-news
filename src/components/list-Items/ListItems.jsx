import React, { useState, useEffect } from 'react';

import './ListItems.scss';

import { Button } from '../../common'

export const ListItems = ({ item }) => {
    const [point, setPoint] = useState(item.points);
    const handlePointClick = (e) => {
        console.log(e.target.current);
        setPoint(point + 1);
        localStorage.setItem(item.objectID,JSON.stringify(point+1))
    }

    useEffect(() => {
       const savedPoint= localStorage ? localStorage[item.objectID] ? JSON.parse(localStorage[item.objectID]): null : null;
        if( savedPoint !== null){
            setPoint(savedPoint);
        }
    },[point, item.objectID]);


    return (
        <li className="display-container">
            <div className="badge-container">
                <span className="badge badge-salmon">{item.num_comments}</span>
                <span className="badge badge-green">{point}</span>
            </div>
            <Button refs={item.objectID} onClick={e=> handlePointClick(e)}>&#9650;</Button>
            {item.title}
            <span className="url">{
                item.url ?
                    <a href={item.url}>
                        {"(" + item.url.split('//')[1].split('/')[0] + ")"}
                    </a>
                    : null}
            </span>
            <span className="btn-display-right"><Button>&times;</Button></span>
        </li>
    );
};