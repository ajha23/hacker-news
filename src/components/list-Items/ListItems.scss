.display-container{
    position: relative;
    background-color: linen;
}
ul li {
    padding:8px 16px;
    border-bottom:1px solid #ddd
}
ul li:last-child {
    border-bottom:none;
}

.badge {
    background-color:black;
    color:#fff;
    display:inline-block;
    padding-left:0.325em;
    padding-right:0.325em;
    text-align:center;
    border-radius:50%;
    margin-left: 6px;
    margin-right: 8px;
}

.badge-salmon {
    background-color: lightsalmon;
}

.badge-green {
    background-color: limegreen;
}

.btn-display-right {
    position: absolute;
    top: 50%;
    right: 0%;
    transform: translate(0%,-50%);
}

.url {
    font-family: Verdana, Geneva, sans-serif;
    font-size: 8pt;
    color: #828282;
    margin-left: .625em;
}

.badge-container {
    display:inline-block; 
    width:20%;
    text-align:center;
}
