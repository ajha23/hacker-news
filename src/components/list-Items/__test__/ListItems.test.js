import React from 'react';
import { render } from '@testing-library/react';
import { ListItems } from '../ListItems';
test('ListItems', () => {
  const item = { title: 'By Amazon', points:671, comments:678 };
  render(<ListItems item={item} />);
  expect(render()).toMatchSnapshot();
});