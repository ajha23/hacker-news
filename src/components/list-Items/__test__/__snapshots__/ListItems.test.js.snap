// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="display-container"
      >
        <div
          class="badge-container"
        >
          <span
            class="badge badge-salmon"
          />
          <span
            class="badge badge-green"
          >
            671
          </span>
        </div>
        <div
          class="title-container"
        >
          <span>
            <span
              class="btn"
            >
              ▲
            </span>
          </span>
          By Amazon
          <span
            class="url"
          />
        </div>
        <span
          class="btn-display-right"
        >
          <span
            class="btn"
          >
            ×
          </span>
        </span>
      </li>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
